lib/web/users_test.go

```diff
@@ -118,6 +119,186 @@ func TestCRUDs(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestSetTrait(t *testing.T) {
+	defaultRoles := []string{"role1"}
+	tests := []struct {
+		name           string
+		updateReq      saveUserRequest
+		expectedTraits map[string][]string
+	}{
+		{
+			name: "Logins",
+			updateReq: saveUserRequest{
+				Name:   "setlogins",
+				Roles:  defaultRoles,
+				Logins: &[]string{"login1", "login2"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitLogins: {"login1", "login2"},
+			},
+		},
+		{
+			name: "DB",
+			updateReq: saveUserRequest{
+				Name:    "setdb",
+				Roles:   defaultRoles,
+				DBUsers: &[]string{"dbuser1", "dbuser2"},
+				DBNames: &[]string{"dbname1", "dbname2"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitDBUsers: {"dbuser1", "dbuser2"},
+				teleport.TraitDBNames: {"dbname1", "dbname2"},
+			},
+		},
+		{
+			name: "Kube",
+			updateReq: saveUserRequest{
+				Name:       "setkube",
+				Roles:      defaultRoles,
+				KubeUsers:  &[]string{"kubeuser1", "kubeuser2"},
+				KubeGroups: &[]string{"kubegroup1", "kubegroup2"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitKubeUsers:  {"kubeuser1", "kubeuser2"},
+				teleport.TraitKubeGroups: {"kubegroup1", "kubegroup2"},
+			},
+		},
+		{
+			name: "WindowsLogins",
+			updateReq: saveUserRequest{
+				Name:          "setwindowslogins",
+				Roles:         defaultRoles,
+				WindowsLogins: &[]string{"login1", "login2"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitWindowsLogins: {"login1", "login2"},
+			},
+		},
+		{
+			name: "AWSRoleARNs",
+			updateReq: saveUserRequest{
+				Name:        "setawsrolearns",
+				Roles:       defaultRoles,
+				AWSRolesARN: &[]string{"arn1", "arn2"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitAWSRoleARNs: {"arn1", "arn2"},
+			},
+		},
+		{
+			name: "Deduplicates",
+			updateReq: saveUserRequest{
+				Name:   "deduplicates",
+				Roles:  defaultRoles,
+				Logins: &[]string{"login1", "login2", "login1"},
+			},
+			expectedTraits: map[string][]string{
+				teleport.TraitLogins: {"login1", "login2"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
```suggestion
		tt := tt
		t.Run(tt.name, func(t *testing.T) {
```

Maybe it's a bit paranoid, but the `t.Parallel()` makes me worry that we could get capture-by-reference issues on `tt`.

(Or drop the Parallel? I don't think it's the panacea that will make our tests Blazing Fast.)